Content.Server/GameObjects/Components/Power/PowerNetComponents/SolarControlConsoleComponent.cs

```diff
@@ -12,26 +12,20 @@
 namespace Content.Server.GameObjects.Components.Power.PowerNetComponents
 {
     [RegisterComponent]
-    [ComponentReference(typeof(IActivate))]
-    public class SolarControlConsoleComponent : SharedSolarControlConsoleComponent, IActivate
+    [ComponentReference(typeof(BaseComputerUserInterfaceComponent))]
+    public class SolarControlConsoleComponent : BaseComputerUserInterfaceComponent
     {
+        public override string Name => "SolarControlConsole";
+
         [Dependency] private readonly IEntitySystemManager _entitySystemManager = default!;
 
         private PowerSolarSystem _powerSolarSystem = default!;
-        private bool Powered => !Owner.TryGetComponent(out PowerReceiverComponent? receiver) || receiver.Powered;
 
-        [ViewVariables] private BoundUserInterface? UserInterface => Owner.GetUIOrNull(SolarControlConsoleUiKey.Key);
+        public SolarControlConsoleComponent() : base(SolarControlConsoleUiKey.Key) { }
```

Comment:
You should not be using constructors like this for components.

Reply:
See other review of this kind.